### PR TITLE
Adds GeoHex grid documentation

### LINKED ISSUE #1100: [DOC] Support aggregation on geo_point based on H3 library
**What do you want to do?**
 
- [x] Add new documentation

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
geo_point field will support new aggregation based on H3 grid system.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/geospatial/issues/98

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #1100 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/geohexgrid-agg.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+## Precision
+
+The `precision` parameter controls the level of granularity that determines the grid cell size. The lower the precision, the larger the grid cells. 
+
+The following example illustrates low-precision and high-precision aggregation requests.
+
+To start, create an index and map the `location` field as geopoint:
+
+```json
+PUT testindex
```
Comment (VijayanB) [inline _opensearch/geohexgrid-agg.md:26]: nit
```suggestion
PUT national_parks
```

**Review by VijayanB — APPROVED** [review]

File: _opensearch/geohexgrid-agg.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+layout: default
+title: GeoHex grid aggregations
+parent: Aggregations
+nav_order: 4
+has_children: false
+---
+
+# GeoHex grid aggregations
+
+The Hexagonal hierarchical geospatial indexing system (H3) partitions the Earth's areas into identifiable hexagon-shaped cells.
```
Comment (vagimeli) [inline _opensearch/geohexgrid-agg.md:11]: This is a funny one. It's a proper name, so it should be written as Hexagonal Hierarchical Geospatial Indexing System (H3). Uber refers to it as H3 on first mention, ignoring capitalization standards for proper names. It's their product though, so they can do what they want. I'd follow capitalization standards for proper names in our documentation.

File: _opensearch/geohexgrid-agg.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+  "location": "36.53, -116.93" 
+}
+```
+
+You can index geopoints in several formats. For a list of all supported formats, see the [geopoint documentation]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats). 
+{: .note}
+
+## Low-precision requests
+
+Run a low-precision request that should bucket all three documents together:
```
Comment (vagimeli) [inline _opensearch/geohexgrid-agg.md:65]: Should this read "..that buckets all..."

File: _opensearch/geohexgrid-agg.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+        }
+      ]
+    }
+  }
+}
+```
+
+## Filtering requests
+
+High-precision requests are resource-intensive, so we recommend to use a filter like `geo_bounding_box` to limit the geographical area. For example, the following query applies a filter to limit the search area:
```
Comment (vagimeli) [inline _opensearch/geohexgrid-agg.md:238]: Suggested revision: "..., so we **recommend using** a filter..."

File: _opensearch/geohexgrid-agg.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+            "top_left": "POINT (-120 38)",
+            "bottom_right": "POINT (-116 36)"
+        }
+      }
+    }
+  }
+}
+```
+
+The response contains only the two results, which are within the specified bounds:
```
Comment (vagimeli) [inline _opensearch/geohexgrid-agg.md:328]: Change "...two results, which" to "two results that"

File: _opensearch/geohexgrid-agg.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+precision | Integer | The zoom level used to determine grid cells for bucketing results. Valid values are in the [0, 15] range. Default is 5. 
+bounds | Object | The bounding box for filtering geopoints. The bounding box is defined by the top left and bottom right vertices. The vertices are specified as geopoints in one of the following formats: <br>- An object with a latitude and longitude<br>- An array in the [`longitude`, `latitude`] format<br>- A string in the "`latitude`,`longitude`" format<br>- A geohash <br>- Well-known text (WKT).<br> See the [geopoint formats]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats) for format examples. Optional.
+size | Integer | The maximum number of buckets to return. When there are more buckets than `size`, OpenSearch returns buckets with more documents. Optional. Default is 10,000.
+shard_size | Integer | The maximum number of buckets to return from each shard. Optional. Default is max (10, `size` &middot; number of shards), which gives a more accurate count of higher prioritized buckets.
```
Comment (vagimeli) [inline _opensearch/geohexgrid-agg.md:377]: Change "higher prioritized" to "higher-prioritized"

**Review by vagimeli — APPROVED** [review]
Minimal edits. Looks good.

**Review by ariamarble — APPROVED** [review]
LGTM other than my comments

File: _opensearch/geohexgrid-agg.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+The `bounds` parameter can be used with or without the `geo_bounding_box` filter; these two parameters are independent and can have any spatial relationship to each other.
+
+## Supported parameters
+
+GeoHex grid aggregation requests support the following parameters.
+
+Parameter | Data Type | Description
+:--- | :--- | :---
+field | String | The field that contains the geopoints. This field must be mapped as a `geo_point` field. If the field contains an array, all array values are aggregated. Required.
+precision | Integer | The zoom level used to determine grid cells for bucketing results. Valid values are in the [0, 15] range. Default is 5.
```
[the remark this answers is not in the document]
Comment (VijayanB) [reply _opensearch/geohexgrid-agg.md:374]: optional

File: _opensearch/geohexgrid-agg.md (hunk 3 above)
Comment (kolchfa-aws) [reply _opensearch/geohexgrid-agg.md:65]: This is what's expected to happen after you run this request.

Comment (natebower) [reply _opensearch/geohexgrid-agg.md:65]: I agree that "should" reads strangely here and would go with @vagimeli suggestion, if accurate.

File: _opensearch/geohexgrid-agg.md [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+      "geohex_grid": {
+        "field": "location",
+        "precision": 1
+      }
+    }
+  }
+}
+```
+
+You can use either `GET` or `POST` HTTP method for GeoHex grid aggregation queries.
```
Comment (natebower) [inline _opensearch/geohexgrid-agg.md:81]: Should this read "You can use either the `GET` or `POST` HTTP method for..."?

File: _opensearch/geohexgrid-agg.md [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+        }
+      ]
+    }
+  }
+}
+```
+
+## High-precision requests
+
+Now run a higher precision request:
```
Comment (natebower) [inline _opensearch/geohexgrid-agg.md:151]:
```suggestion
Now run a high-precision request:
```

File: _opensearch/geohexgrid-agg.md [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+        }
+      ]
+    }
+  }
+}
+```
+
+## Filtering requests
+
+High-precision requests are resource-intensive, so we recommend using a filter like `geo_bounding_box` to limit the geographical area. For example, the following query applies a filter to limit the search area:
```
Comment (natebower) [inline _opensearch/geohexgrid-agg.md:238]:
```suggestion
High-precision requests are resource intensive, so we recommend using a filter like `geo_bounding_box` to limit the geographical area. For example, the following query applies a filter to limit the search area:
```

Comment (natebower) [inline _opensearch/geohexgrid-agg.md:238]: Compound modifiers only need to be hyphenated when they precede the noun they modify. So it's a resource-intensive request, but the request is resource intensive.

File: _opensearch/geohexgrid-agg.md [hunk 11]
```diff
@@ -0,0 +1,377 @@
...
+            "doc_count" : 1
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+You can also restrict the geographical area by providing the coordinates of the bounding envelope in the `bounds` parameter. Both `bounds` and `geo_bounding_box` coordinates can be specified in any of the [geopoint formats]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats). The following query uses the WKT "POINT(`longitude` `latitude`)" format for the `bounds` parameter:
```
Comment (natebower) [inline _opensearch/geohexgrid-agg.md:307]:
```suggestion
You can also restrict the geographical area by providing the coordinates of the bounding envelope in the `bounds` parameter. Both `bounds` and `geo_bounding_box` coordinates can be specified in any of the [geopoint formats]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats). The following query uses the well-known text (WKT) "POINT(`longitude` `latitude`)" format for the `bounds` parameter:
```

File: _opensearch/geohexgrid-agg.md [hunk 12]
```diff
@@ -0,0 +1,377 @@
...
+
+## Supported parameters
+
+GeoHex grid aggregation requests support the following parameters.
+
+Parameter | Data Type | Description
+:--- | :--- | :---
+field | String | The field that contains the geopoints. This field must be mapped as a `geo_point` field. If the field contains an array, all array values are aggregated. Required.
+precision | Integer | The zoom level used to determine grid cells for bucketing results. Valid values are in the [0, 15] range. Optional. Default is 5. 
+bounds | Object | The bounding box for filtering geopoints. The bounding box is defined by the top left and bottom right vertices. The vertices are specified as geopoints in one of the following formats: <br>- An object with a latitude and longitude<br>- An array in the [`longitude`, `latitude`] format<br>- A string in the "`latitude`,`longitude`" format<br>- A geohash <br>- Well-known text (WKT).<br> See the [geopoint formats]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats) for format examples. Optional.
```
Comment (natebower) [inline _opensearch/geohexgrid-agg.md:375]:
```suggestion
bounds | Object | The bounding box for filtering geopoints. The bounding box is defined by the top left and bottom right vertices. The vertices are specified as geopoints in one of the following formats: <br>- An object with a latitude and longitude<br>- An array in the [`longitude`, `latitude`] format<br>- A string in the "`latitude`,`longitude`" format<br>- A geohash <br>- WKT<br> See the [geopoint formats]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats) for format examples. Optional.
```

Comment (natebower) [inline _opensearch/geohexgrid-agg.md:375]: Should "geohash" be capitalized? "for formatting examples"?

File: _opensearch/geohexgrid-agg.md [hunk 13]
```diff
@@ -0,0 +1,377 @@
...
+size | Integer | The maximum number of buckets to return. When there are more buckets than `size`, OpenSearch returns buckets with more documents. Optional. Default is 10,000.
+shard_size | Integer | The maximum number of buckets to return from each shard. Optional. Default is max (10, `size` &middot; number of shards), which gives a more accurate count of higher-prioritized buckets.
```
Comment (natebower) [inline _opensearch/geohexgrid-agg.md:377]:
```suggestion
shard_size | Integer | The maximum number of buckets to return from each shard. Optional. Default is max (10, `size` &middot; number of shards), which provides a more accurate count of higher-prioritized buckets.
```

Comment (natebower) [inline _opensearch/geohexgrid-agg.md:377]: "more highly prioritized buckets"?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions.

File: _opensearch/geohexgrid-agg.md [hunk 14]
```diff
@@ -0,0 +1,377 @@
...
+parent: Aggregations
+nav_order: 4
+has_children: false
+---
+
+# GeoHex grid aggregations
+
+The Hexagonal Hierarchical Geospatial Indexing System (H3) partitions the Earth's areas into identifiable hexagon-shaped cells. 
+
+The H3 grid system works well for proximity applications, because it overcomes the shortcomings of geohash's non-uniform partitions. Geohash encodes latitude and longitude pairs, leading to significantly smaller partitions near the poles and a degree of longitude near the equator. However, the H3 grid system's distortions are low and limited to five partitions of 122. These five partitions are placed in low-use areas (for example, in the middle of the ocean), leaving the essential areas error-free. Thus, grouping documents based on the H3 grid system provides a better aggregation than the geohash grid.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _opensearch/geohexgrid-agg.md:13]: Replaced with "overcomes the limitations". Geohash is lowercase in general.

[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _opensearch/geohexgrid-agg.md:13]: Just checked, and Geohash is capitalized. Will change.

**Pull request merged**